frappe/custom/doctype/custom_field/custom_field.py

```diff
@@ -234,6 +234,21 @@ def on_trash(self):
 					layout_doc.save()
 					break
 
+		if not frappe.flags.in_migrate:
+			# Check if there are more than one distinct non-null values.
+			# If not, it's safe to delete the column.
+			if (
+				frappe.db.has_column(self.dt, self.fieldname)
+				and len(frappe.db.get_all(self.dt, pluck=self.fieldname, distinct=True, limit=2)) < 2
+			):
+				delete_fields({self.dt: [self.fieldname]}, delete=1)
+			else:
+				frappe.throw(
```

Comment:
This is breaking change :sweat_smile: let's offer a separate way to delete the field regardless of data?

Reply:
Another solution I could image is to have a confirmation dialog in Customize Form, letting the use know that data will be lost and if they want to (a) undo their changes, (b) drop the column from schema and DB, or (c) drop the column from schema only.

But my feeling is that the last option will remain mostly unused.

Reply:
> Just pretending to delete it and providing a separate UI to delete it would be a bad UX, IMO.

Bad UX > irrecoverable data loss.


To improve UX, What we can do is prompt the user here that `custom field is deleted, do you also want to delete column completely from database?`

Reply:
Also users might need to delete columns with >1 distinct values. Blocking that completely doesn't seem like a good idea.